packages/cli/src/commands/template/build.ts

```diff
@@ -261,47 +264,53 @@ export const buildCommand = new commander.Command('build')
         )
 
         try {
-          child_process.execSync(
-            `echo "${accessToken}" | docker login docker.${e2b.SANDBOX_DOMAIN} -u _e2b_access_token --password-stdin`,
-            {
-              stdio: 'inherit',
-              cwd: root,
-            },
-          )
+          docker.checkAuth({
+            username: '_e2b_access_token',
+            password: accessToken,
+            serveraddress: `docker.${e2b.SANDBOX_DOMAIN}`,
+          })
         } catch (err: any) {
           console.error(
-            'Docker login failed. Please try to login with `e2b auth login` and try again.',
+            'Docker login failed. Please try to login with `e2b auth login` and try again.'
           )
           process.exit(1)
         }
+
         process.stdout.write('\n')
 
         console.log('Building docker image...')
-        const cmd = `docker build . -f ${dockerfileRelativePath} --platform linux/amd64 -t docker.${
-          e2b.SANDBOX_DOMAIN
-        }/e2b/custom-envs/${templateID}:${template.buildID} ${Object.entries(
-          dockerBuildArgs,
-        )
-          .map(([key, value]) => `--build-arg="${key}=${value}"`)
-          .join(' ')}`
-        child_process.execSync(cmd, {
-          stdio: 'inherit',
-          cwd: root,
-          env: {
-            ...process.env,
-            DOCKER_CLI_HINTS: 'false',
-          },
+
+        const dockerImageTag = `docker.${e2b.SANDBOX_DOMAIN}/e2b/custom-envs/${templateID}:${template.buildID}`
+        const buildStream = await docker.buildImage(await prepareDockerContext(root), {
+          t: dockerImageTag,
+          platform: 'linux/amd64',
+          buildargs: dockerBuildArgs,
+          dockerfile: dockerfileRelativePath,
         })
```

Comment:
After some brief tests (dockerignore, nested dir in context) I think it might be possible to just pass root and list of top level files (from https://github.com/apocas/dockerode/blob/49bd8c4c231efe9bb432b28d51dbae3e50ddd010/test/docker.js#L174 it looks like they are already handling this internally) instead of using tar-fs + filtering.

```suggestion
        const buildStream = await docker.buildImage({
          context: root,
          src: fs.readdirSync(root),
        }, {
          t: dockerImageTag,
          platform: 'linux/amd64',
          buildargs: dockerBuildArgs,
          dockerfile: dockerfileRelativePath,
        })
```